### PR TITLE
Move linked cluster tests to adapter

### PR DESCRIPTION
**Pull request by umanwizard**
This allows us to avoid the check in the case where we are modifying builtin introspection sources.

### CODE REVIEW

File: src/adapter/src/coord/sequencer.rs [hunk 1]
```diff
@@ -1459,6 +1488,18 @@ impl Coordinator {
             if_not_exists,
         } = plan;
 
+        if !self
+            .catalog
+            .state()
+            .is_system_schema_specifier(&name.qualifiers.schema_spec)
```
Comment (benesch) [inline src/adapter/src/coord/sequencer.rs:1494]: Wait, this is so weird. We don't use _system_ IDs for these indexes? Huh.

File: src/adapter/src/coord/sequencer.rs [hunk 2]
```diff
@@ -471,6 +472,32 @@ impl Coordinator {
         }
     }
 
+    // Utilities used by `sequence` functions
+    // ---
+    fn ensure_cluster_is_not_linked(&self, cluster_id: ClusterId) -> Result<(), AdapterError> {
+        if let Some(linked_id) = self.catalog.get_linked_object(cluster_id) {
```
Comment (benesch) [inline src/adapter/src/coord/sequencer.rs:478]: I think `get_linked_object` is not worth it's weight! Given that you then have to fetch the cluster below. Can we instead do:

```
let cluster = self.catalog.get_cluster(cluster_id);
if let Some(linked_id) = cluster.linked_object_id {
    let cluster_name = cluster.name.clone();
}
```

File: src/adapter/src/coord/sequencer.rs [hunk 3]
```diff
@@ -471,6 +472,32 @@ impl Coordinator {
         }
     }
 
+    // Utilities used by `sequence` functions
```
Comment (benesch) [inline src/adapter/src/coord/sequencer.rs:475]: This seems a little weird. We've got a bunch of utility funcs at the end of the `impl` block, if you instead wanted to toss this down there?

File: src/adapter/src/coord/sequencer.rs (hunk 1 above)
Comment (umanwizard) [reply src/adapter/src/coord/sequencer.rs:1494]: Yep.

**Comment by umanwizard** [comment]
I have made the changes @benesch requested.

**Review by benesch — APPROVED** [review]

**Review by jkosh44 — APPROVED** [review]
LGTM, but is there anyway to test if it actually worked?

**Comment by umanwizard** [comment]
Yes... my `sum` PR that will go on top of it immediately after. That's why
I didn't think it was crucial to add tests here.

**Pull request merged**